### PR TITLE
unsubscribe. tests

### PR DESCRIPTION
**Pull request by mateigraura**

### CODE REVIEW

File: dispatcher/hub/commonHub.go [hunk 1]
```diff
@@ -99,18 +98,22 @@ func (wh *commonHub) handleBroadcast(events []data.Event) {
...
 	wh.dispatchers[d.GetID()] = d
 }
 
 func (wh *commonHub) unregisterDispatcher(d dispatcher.EventDispatcher) {
 	wh.rwMut.Lock()
-	defer wh.rwMut.Unlock()
 	if _, ok := wh.dispatchers[d.GetID()]; ok {
 		delete(wh.dispatchers, d.GetID())
 	}
+	wh.rwMut.Unlock()
```
Comment (claudiulataretu) [inline dispatcher/hub/commonHub.go:116]: why did you not use it as a defer like on registerDispatched?

File: dispatcher/subscription.go [hunk 2]
```diff
@@ -77,26 +77,42 @@ func (sm *SubscriptionMap) MatchSubscribeEvent(event SubscribeEvent) {
...
+func (sm *SubscriptionMapper) matchLevelFromInput(subEntry SubscriptionEntry) string {
 	hasAddress := subEntry.Address != "" && strings.Contains(subEntry.Address, erdTag)
 	hasIdentifier := subEntry.Identifier != ""
 	hasTopics := len(subEntry.Topics) > 0
 
 	if hasAddress && hasIdentifier && hasTopics {
 		return MatchTopics
 	}
-	if hasAddress && hasIdentifier {
+	if hasAddress && hasIdentifier || hasIdentifier{
```
Comment (claudiulataretu) [inline dispatcher/subscription.go:115]: I'm not sure you need hasAddress. If hasIdentifier is false then if expression will be false. Also if hasAddress is false and hasIdentifier is true then if expression is true. This means that only hasIdentifier will have an impact on the if resolve

File: dispatcher/hub/commonHub.go (hunk 1 above)
Comment (mateigraura) [reply dispatcher/hub/commonHub.go]: fixed

File: dispatcher/subscription.go (hunk 2 above)
Comment (mateigraura) [reply dispatcher/subscription.go]: done. added new matchLevel

**Review by catalinnnn — APPROVED** [review]

**Review by claudiulataretu — APPROVED** [review]

**Pull request merged**